datumaro/util/mask_tools.py

```diff
@@ -64,7 +64,7 @@ def unpaint_mask(painted_mask, inverse_colormap=None):
     if callable(inverse_colormap):
         map_fn = lambda a: inverse_colormap((a >> 16) & 255, (a >> 8) & 255, a & 255)
     else:
-        map_fn = lambda a: inverse_colormap[((a >> 16) & 255, (a >> 8) & 255, a & 255)]
+        map_fn = lambda a: inverse_colormap.get(((a >> 16) & 255, (a >> 8) & 255, a & 255), 0)
```

Comment:
Can you explain this change?

Reply:
While testing this format on some dataset, I found this problem. The masks have pixels that could not be identified using the given classes. Such pixels should be defined as background.

Reply:
> Such pixels should be defined as background.

Where this comes from? Why not throw an error?

Also, we don't always have background at id 0 already, so maybe it should be a parameter.

Reply:
I think that if a pixel does not belong to any presented class, it belongs to the background (it seemed to me that this class was partially needed for this). But I agree that this can be interpreted as an error in the dataset.

> Also, we don't always have background at id 0 already, so maybe it should be a parameter.

Do you mean add a parameter to the `load_mask` function for the background class number?

Reply:
+1 to treating this as an error.

Reply:
Added an error for this case.

Reply:
> I think that if a pixel does not belong to any presented class, it belongs to the background 

Why?

> it seemed to me that this class was partially needed for this. But I agree that this can be interpreted as an error in the dataset.

I agree, that it can be true in some cases, like id 255 / color 255,255,255 in VOC masks is used for "boundary" pixels, which are not supposed to be used for accuracy evaluation. But this fact cannot be treated as a common practice, or always supposed.

> Do you mean add a parameter to the load_mask function for the background class number?

No, I'm speaking about the `unpaint_mask` function, which is being discussed in this thread. We could add a parameter to map unknown pixels to background, if it is needed, but it should not be the default behavior.